### PR TITLE
Pretty print JSON, add type names.

### PR DESCRIPTION
**Pull request by oskidan**
Hi everyone, :wave:

Hopefully this PR makes the generated JSON a bit more readable.

Resolves: #70

### CODE REVIEW

File: Sources/Core/AST/AnyNode.swift [hunk 1]
```diff
@@ -27,8 +27,9 @@ struct AnyNode: Codable {
   func encode(to encoder: Encoder) throws {
     var container = encoder.container(keyedBy: CodingKeys.self)
 
-    let kind = type(of: node).kind
-    try container.encode(kind, forKey: .kind)
+    let nodeType = type(of: node)
+    try container.encode("\(nodeType)", forKey: .nodeName)
```
Comment (kyouko-taiga) [inline Sources/Core/AST/AnyNode.swift:31]:
```suggestion
    let nodeType = type(of: node)
    try container.encode("\(type(of: node))", forKey: .nodeName)
```

File: Sources/Core/AST/AnyNode.swift [hunk 2]
```diff
@@ -5,7 +5,7 @@ struct AnyNode: Codable {
 
   fileprivate enum CodingKeys: String, CodingKey {
 
-    case kind, data
+    case nodeName, kind, data
```
Comment (kyouko-taiga) [inline Sources/Core/AST/AnyNode.swift:8]:
```suggestion
    case nodeType, kind, data
```

Comment (kyouko-taiga) [inline Sources/Core/AST/AnyNode.swift:8]:
```suggestion
    case nodeType, kind, data
```

**Review by kyouko-taiga — CHANGES_REQUESTED** [review]
Thanks for this contribution!

I think `CodingKeys.nodeName` should be renamed `CodingKeys.nodeType`. Except that, I the changes should make json outputs more usable. Good job.

File: Sources/Core/AST/AnyNode.swift (hunk 2 above)
Comment (oskidan) [reply Sources/Core/AST/AnyNode.swift]: Makes sense. Done.

**Review by kyouko-taiga — APPROVED** [review]
LGTM

**Review by dabrahams — CHANGES_REQUESTED** [review]
Sorry, I can't tell what this accomplishes by looking at the code.  I'd like to see a test added that would fail before these changes and passes afterward.  Thanks!

**Comment by C-BJ** [comment]
What's wrong with this PR? @kyouko-taiga

**Comment by kyouko-taiga** [comment]
Dave asked for a test that would highlight what the PR is accomplishing.

**Comment by kyouko-taiga** [comment]
I think this PR has gone stale.

There's a related issue on the topic of visualizing ASTs (see #70).

**Pull request closed without merging**